### PR TITLE
Save the sending emails to the database

### LINKED ISSUE #325: Implement a mechanism to save the sending emails to the database
**Definition of done**
There is a table in the database the administrators can check as an email log that contains all of the automatic and custom emails sent through the platform 

**Addition information**
In addition to the common data attributes of the email, 
- The program id
- The program state in which the email was sent
- The time it was sent out

Should also be recorded

### PR DESCRIPTION
**Pull request by YaiyaSam**
## Purpose

The purpose of this PR is to fix #325

## Goals

It is required to keep a log of every email sent out by the system, in the database so it can be easily accessed by the admins whenever needed. 

### Screenshots

image

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/main/java/org/sefglobal/scholarx/model/SentEmails.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        return receiver;
+    }
+
+    // public String getUserType() {
+    //     return userType;
+    // }
+
+    // public void setUserType(String userType) {
+    //     this.userType = userType;
+    // }
```
Comment (Gravewalker666) [inline src/main/java/org/sefglobal/scholarx/model/SentEmails.java:77]: lets remove these commented lines

File: src/main/java/org/sefglobal/scholarx/model/SentEmails.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import javax.persistence.ManyToOne;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.sefglobal.scholarx.util.ProgramState;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+ 
+@Entity
+@Table(name = "email")
+@JsonIgnoreProperties({"createdAt"})
+public class SentEmails extends BaseScholarxModel {
```
Comment (Gravewalker666) [inline src/main/java/org/sefglobal/scholarx/model/SentEmails.java:15]:
```suggestion
public class SentEmail extends BaseScholarxModel {
```

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]
Amazing work @YaiyaSam 
👏 👏 👏 

Some minor changes requested, but overall I'm happy with the solution you have given for the pretty vague requirement we gave, giving you the space to come up with a solution of your own.

File: src/main/java/org/sefglobal/scholarx/model/SentEmails.java (hunk 2 above)
Comment (YaiyaSam) [reply src/main/java/org/sefglobal/scholarx/model/SentEmails.java]: Fixed it with commit d8de1214775427ed2572e95fd6f81f526778a504

**Review by Gravewalker666 — APPROVED** [review]

**Pull request merged**